crates/sui-framework/sources/coin.move

```diff
@@ -234,6 +271,43 @@ module sui::coin {
         burn(c, coin);
     }
 
+    // === Update coin metadata ===
+
+    /// Update decimals in `CoinMetadata`
+    public entry fun update_decimals<T>(
```

Comment:
This update function in particular seems security-sensitive since some DeFi code tends to look at `decimals` for conversion purposes. My preference would be to make this one immutable out of caution and have the security vs convenience discussion if someone expresses a strong need later.